.github/workflows/check-file-updates.yaml

```diff
@@ -0,0 +1,65 @@
+name: Check config and readme updates
+on:
+  pull_request:
+    paths:
+      - "apis/**"
+jobs:
+  file-updates:
+    permissions: write-all
+    name: Test file updates
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run make commands
+        run: |
+          make generate && make manifests && make api-docs
```

Comment:
why do we need  `generate` and `manifests` here?

Reply:
is this change to cover more than just api doc but also  other files?

Reply:
so it is to cover the changes made in the crd as well

Reply:
basically it is to update any auto-generated files

Reply:
Isn't it in general overcomplicated? Should it be enough just fails if git status shows uncommitted changes after the make targets run? All the details easy to get from logs and running make locally.

Reply:
😅  Just wanted to provide the information very easily

Reply:
Do you have any plans to change linter check where the problems are really on per-file basis and no details in the logs?

Reply:
for linter, could we do something here -> https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/.github/workflows/linter.yaml

Reply:
> Do you have any plans to change linter check where the problems are really on per-file basis and no details in the logs?

I [wanted to](https://github.com/opendatahub-io/opendatahub-operator/pull/864#discussion_r1489064260) but seemingly #865 didn't fix it. I wonder if we are missing some scopes in the token we use, but I don't have access to these settings to verify. That would be both per-file and logs though.

Personally, I do not need either of those jobs, as I made it part of my "pre-push" hook, but it is good to have them automated so we don't need to bother about it when reviewing the PR.

Reply:
regarding linters, I will try to fix it in a separate pr? sounds good?

Reply:
I will take a look for sure :+1